### ISSUE TITLE
AttributeError: 'list' object has no attribute 'lower'

### ISSUE THREAD

**Issue by NicolasLM** [mozilla/bleach#398]
I upgraded to bleach 3.0.0 and noticed an error that happens from time to time: 

```
AttributeError: 'list' object has no attribute 'lower'
  File "bleach/__init__.py", line 84, in clean
    return cleaner.clean(text)
  File "bleach/sanitizer.py", line 169, in clean
    dom = self.parser.parseFragment(text)
  File "bleach/_vendor/html5lib/html5parser.py", line 317, in parseFragment
    self._parse(stream, True, *args, **kwargs)
  File "bleach/html5lib_shim.py", line 257, in _parse
    self.mainLoop()
  File "bleach/_vendor/html5lib/html5parser.py", line 205, in mainLoop
    for token in self.normalizedTokens():
  File "bleach/_vendor/html5lib/html5parser.py", line 264, in normalizedTokens
    for token in self.tokenizer:
  File "bleach/html5lib_shim.py", line 130, in __iter__
    token_name = token['data'].lower().strip()
```

https://github.com/mozilla/bleach/blob/344df3acd1a52953a6006e9a7e066eabd614d28f/bleach/html5lib_shim.py#L130

Here is the value of `token`:
```python
token = {'data': [['"', ''], ['alt', ''], ['class', 'alignnone'], ['height', '100%'], ['src', 'https://foo.com/media/bar.jpg'], ['style', 'max-width: 100%; margin-left: auto; margin-right: auto;'], ['width', '100%']], 'name': 'img', 'selfClosing': True, 'selfClosingAcknowledged': False, 'type': 3}
```

Note that I have `html5lib==1.0.1` installed.

**Comment by willkg** [comment]
@NicolasLM I'm having difficulties creating a failing test case based on the information in this issue. Can you elucidate what it is you're cleaning and how it's being called?

**Comment by NicolasLM** [comment]
@willkg You are right this bug report is a bit light, Sentry did not let me see the full string so I had to dig a bit deeper.

Here you can see how I call `bleach.clean`: https://github.com/NicolasLM/feedsubs/blob/7da4e721c520426d81cc95938d139c4815c8d7bc/reader/html_processing.py#L37

Here is the repr of a failing string (it comes from the [StackOverflow RSS feed](http://careers.stackoverflow.com/jobs/feed?allowsremote=True)):
```python
"<p>\n <span>\n  At Big Nerd Ranch we empower individuals and organizations to dream big and achieve more through innovative digital product development and training. \xa0We do this through our signature immersive bootcamps, our quality online trainings, our elite consulting capabilities and our best selling\xa0Big Nerd Ranch Guides.\n </span>\n</p>\n<br/>\n<br/>\n<p>\n <span>\n  We have a vision to build relationships that drive partnerships with 20 Fortune 500 companies and train 200,000 designers and engineers by 2020. \xa0How are we going to do it? \xa0It's all our people -- we only hire the best. \xa0We are passionate about writing stylish code and teaching others to do the same. \xa0We then live through our values. \xa0We are a\xa0community\xa0of Nerds who are Brilliant and Curious, Hard-working and Accountable, Kind and Authentic.\n </span>\n</p>\n<br/>\n<br/>\n<p>\n <span>\n  Sound appealing? Learn more about us.\n </span>\n</p>\n<br/>\n<br/>\n<p>\n <span>\n  <strong>\n   As a Developer, you will...\n  </strong>\n </span>\n</p>\n<br/>\n<br/>\n<ul>\n <br/>\n <li>\n  <span>\n   Learn every single day\n  </span>\n </li>\n <br/>\n <li>\n  <span>\n   Have empathy for and communicate clearly with your coworkers and clients\n  </span>\n </li>\n <br/>\n <li>\n  <span>\n   Develop existing and new mobile applications for Big Nerd Ranch and for our clients\n  </span>\n </li>\n <br/>\n <li>\n  <span>\n   Author stories (use cases, technical and functional requirements) and tests\n  </span>\n </li>\n <br/>\n <li>\n  <span>\n   Contribute to and participate in development meetings (standÃ\x82Â\xadups, kickÃ\x82Â\xadoffs, retrospectives) and design sessions with teams and clients\n  </span>\n </li>\n <br/>\n <li>\n  <span>\n   Collaborate with team members, consultants, designers and clients to provide accurate estimations, timelines and risks associated with development\n  </span>\n </li>\n <br/>\n <li>\n  <span>\n   Contribute to the Android community through code, blog posts, teaching and speaking\n  </span>\n </li>\n <br/>\n</ul>\n<br/>\n<br/>\n<p>\n <span>\n  <strong>\n   As a candidate, you...\n  </strong>\n </span>\n</p>\n<br/>\n<br/>\n<ul>\n <br/>\n <li>\n  <span>\n   Have Android development experience and experience working on consulting projects\n  </span>\n </li>\n <br/>\n <li>\n  <span>\n   Understand that an activity is destroyed across a configuration change, but also why it's destroyed and how to embrace its lifecycle\n  </span>\n </li>\n <br/>\n <li>\n  <span>\n   Know how to gracefully support any screen size and device type\n  </span>\n </li>\n <br/>\n <li>\n  <span>\n   Appreciate the quality guarantees that wellÃ\x82Â\xad-written tests provide\n  </span>\n </li>\n <br/>\n <li>\n  <span>\n   Have a drive to learn and understand as much as possible\n  </span>\n </li>\n <br/>\n <li>\n  <span>\n   Enjoy reviewing others' code and having your code reviewed\n  </span>\n </li>\n <br/>\n <li>\n  <span>\n   Appreciate the perspective that others from a variety of different backgrounds and experience levels can provide\n  </span>\n </li>\n <br/>\n <li>\n  <span>\n   Have strong written and verbal communication skills\n  </span>\n </li>\n <br/>\n</ul>\n<br/>\n<br/>\n<p>\n <span>\n  <strong>\n   You might also...\n  </strong>\n </span>\n</p>\n<br/>\n<br/>\n<ul>\n <br/>\n <li>\n  <span>\n   Have experience with Kotlin (if not, don't worry. We'll teach you)\n  </span>\n </li>\n <br/>\n <li>\n  <span>\n   Have knowledge of RxJava and its impact on application architecture\n  </span>\n </li>\n <br/>\n <li>\n  <span>\n   Enjoy teaching others and can clearly describe complicated software to your fellow developers\n  </span>\n </li>\n <br/>\n</ul>\n<br/>\n<br/>\n<p>\n <span>\n  <strong>\n   The benefits we offer to full time employees:\n  </strong>\n </span>\n</p>\n<br/>\n<br/>\n<ul>\n <br/>\n <li>\n  <span>\n   Health, dental, vision, life insurance; 401k matching, short and long term disability\n  </span>\n </li>\n <br/>\n <li>\n  <span>\n   Any hardware, software or services you need to do your job are always 100% on us.\n  </span>\n </li>\n <br/>\n <li>\n  <span>\n   A flexible, remote-friendly and learning based environment where creative ideas are explored and the desire to learn new technologies is encouraged and valued.\n  </span>\n </li>\n <br/>\n <li>\n  <span>\n   Must reside in the US and be legally authorized to work in the US.\n  </span>\n </li>\n <br/>\n</ul>\n<br/>\n<br/>\n<p>\n <span>\n  At\xa0Big Nerd Ranch you will make a difference\xa0and contribute to the development community in interesting ways. \xa0Are you up to the challenge?\n </span>\n</p>"
```

Note that this does not fail with bleach 2.1.4.

**Comment by willkg** [comment]
That was super helpful and I can reproduce it. There's invalid codepoints in the text so that triggers a parsing error and then that causes problems. I'll fix it for 3.0.1.
